components/browser/engine-system/src/main/java/mozilla/components/browser/engine/system/SystemEngineView.kt

```diff
@@ -98,7 +98,7 @@ class SystemEngineView @JvmOverloads constructor(
 
     override fun onDestroy() {
         session?.apply {
-            webView.destroy()
+            (webView.parent as? SystemEngineView)?.removeView(webView)
```

Comment:
I noticed we also call `removeView` in SystemEngineView.render:
https://github.com/mozilla-mobile/android-components/blob/a58f4da8606770094e59ba5e70fcce9fef0dbdc6/components/browser/engine-system/src/main/java/mozilla/components/browser/engine/system/SystemEngineView.kt#L76

e.g. when switching tabs. I don't think there is, but maybe there's a way to consolidate these calls?

Reply:
Yeah, didn't find one that's a big improvement, but we can address if we look into #1945 .